src/core_ocean/shared/mpas_ocn_vmix_cvmix.F

```diff
@@ -668,7 +670,7 @@ subroutine ocn_vmix_coefs_cvmix_build(meshPool, statePool, forcingPool, err, tim
             ! intent out of BoundaryLayerDepth is boundary layer depth measured in meters and vertical index
             indexBoundaryLayerDepth(iCell) = cvmix_variables % kOBL_depth
 
-            do k = 1, maxLevelCell(iCell) + 1
+            do k = minLevelCell(iCell), maxLevelCell(iCell) + 1
```

Comment:
I'm not sure this will work as expected for minLevelCell not equal to 1.  I could imagine since you declare `cvmix_variables % Mdiff_iface` to be nVertLevels+1, you will get the first values in the area entered by cvmix, but with this change you will start at minLevelCell(iCell).  You may need a different counter for that array access below

Reply:
@vanroekel Can you say a little more about what you mean? The strategy I implemented was to keep a correspondence between k-levels in cvmix%* and the other variables in this routine. Are you suggesting it would be better to shift the k-levels in cvmix%* so that they start at minLevelCell?

Reply:
yes of course.  I think the issue is that in the `cvmix_coeffs_kpp` call you pass things like
```
cvmix_variables % Mdiff_iface(minLevelCell(iCell):maxLevelCell(iCell)+1)
```
but have `max_nlev = nVertLevels`.  I think what will happen is that cvmix will be expecting an array that is nVertLevels long and if minLevelCell is not 1 the lengths are inconsistent.  I think you may trip an error in that case or perhaps cvmix will just fill the first maxLevel - minLevel + 1 slots of the array.  I doubt it will fill the correct way.  

Thinking about this, I think the best fix is to change `max_nlev` in L666 to `maxLevelCell(icell) - minLevelCell(iCell) + 1`. I think this will then make the loop variables right here. 

Does that make sense?

Reply:
looking at the code I think we also want `nlev = maxLevelCell(iCell) - minLevelCell(iCell) + 1` on L665

Reply:
I'll try changing `nlev` as you suggest. However, I do wonder if we actually need `max_nlev = maxLevelCell(iCell) - minLevelCell(iCell)+2` since the array limits of the other inputs are `minLevelCell(iCell):maxLevelCell(iCell)+1`

Reply:
oh yes, oops.  You are completely right.  It should be +2, i think they should both be the same.

Reply:
@vanroekel I kept the number of vertical levels specified in arguments to cvmix consistent with `ocean/develop` in order to pass WC tests (https://github.com/MPAS-Dev/MPAS-Model/pull/840/commits/f9e02c36acd725c76cb8acdde8bc5949642d5728).

Reply:
@cbegeman very interesting, I'm not sure why that would be, but definitely not your job to figure out.  If the current iteration passes WC testing, I'll be happy to approve as is.